### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by fourhundredfour**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  * [Branching Model](#branching-model)
+* [Testing](#testing)
+* [Pull Requests](#pull-requests)
+  * [Labels](#labels)
+  * [Reviewer](#reviewer)
+  * [Assignee](#assignee)
+* [Questions](#questions)
+
+## Style Guide
+To keep our project **consistent** and **clean**, every contributor **MUST** following the rules and guidelines.
```
Comment (MargretRensing) [inline CONTRIBUTING.md]:
```suggestion
To keep our project **consistent** and **clean**, every contributor **MUST** follow the rules and guidelines.
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+### Coding Standards
+The coding standards/conventions are usually defined in the README.md. If they do not exist, these fallback rules will be apply:
```
Comment (MargretRensing) [inline CONTRIBUTING.md]:
```suggestion
The coding standards/conventions are usually defined in the README.md. If they do not exist, these fallback rules will apply:
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+#### Default
+* Comments **MUST** be written in **English**.
+* The names of **structures**, **individual types**, **enums**, **classes**, **methods**, **variables**, **functions**, **properties** and **namespaces** **MUST** be written in **English**.
+* Magic Numbers **SHOULD NOT** be used. 
+* The Name of a function or method **SHOULD** be descriptive.
+* Basically this principles **SHOULD** be used: Single Responsibility Principle, Open-Closed Principle, Liskov Substitution Principle, Interface Segregation Principle, Dependency Inversion Principle, YAGNI and KISS.
```
Comment (MargretRensing) [inline CONTRIBUTING.md]:
```suggestion
* Basically these principles **SHOULD** be used: Single Responsibility Principle, Open-Closed Principle, Liskov Substitution Principle, Interface Segregation Principle, Dependency Inversion Principle, YAGNI and KISS.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+#### PHP
+Since other people have already thought about guidelines, the proposal is to adopt the guidelines from [PSR-12](https://www.php-fig.org/psr/psr-12/). [PSR-12](https://www.php-fig.org/psr/psr-12/) complements [PSR-1](https://www.php-fig.org/psr/psr-1/) and [PSR-2](https://www.php-fig.org/psr/psr-2/), so these two **MUST** also be considered.
+In addition, the following rules **MUST** be observed:
+* Classes **MUST** be written in **PascalCase** and set up according to [PSR-4](https://www.php-fig.org/psr/psr-4/) to ensure autoloading.
+* Variable names and Method names **MUST** be written in **camelCase**.
```
Comment (MargretRensing) [inline CONTRIBUTING.md]:
```suggestion
* Variable names and method names **MUST** be written in **camelCase**.
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+### Branching Model
+We use for our project the [Git Flow](https://nvie.com/posts/a-successful-git-branching-model/) Model which is introduced by Vincent Driessen.
```
Comment (MargretRensing) [inline CONTRIBUTING.md]:
```suggestion
For our project we use the [Git Flow](https://nvie.com/posts/a-successful-git-branching-model/) Model which is introduced by Vincent Driessen.
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+We extended the branching model by separating bugfixes from features.
+To commit a bugfix you **MUST** use the new branch type called **Bugfix** which is indicated by the prefix *bugfix/*.
+
+## Testing
+To avoid unexpected behavior of the software we maintain and use automated tests.
+* Every change or implementation **MUST NOT** break any tests. When a test fails, the test **MUST** be adjusted.
```
Comment (MargretRensing) [inline CONTRIBUTING.md]: The test must be adjusted or the code letting the test fail.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+* Every implementation of a new feature **MUST** have at least a unit test.
```
Comment (MargretRensing) [inline CONTRIBUTING.md]:
```suggestion
* Every implementation of a new feature **MUST** have at least one unit test.
```

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+
+## Pull Requests
+Each pull request **SHOULD** have the develop branch as it target branch and should conform the requirements described above.
+
+### Labels
+To sort your pull request correctly, it **MUST** use the available labels.
+**Bug** - Use this label if you have a bugfix or bug report.
+**Documentation** - Use this label if you have a improvement or addition to documentation.
+**Duplicate** - This label is used by the members of the organization to identify duplicates.
+**Feature** - Use this label if you have a new feature or a idea for a new feature.
```
Comment (MargretRensing) [inline CONTRIBUTING.md]:
```suggestion
**Feature** - Use this label if you have a new feature or an idea for a new feature.
```

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+**Question** - Use this label if you have a question.
+
+### Reviewer
+Reviewers will be automatically assigned to your pull request.
+You can put your feet up and wait for your pull request to be accepted ;)
+
+## Questions
+If you have questions about one of our open-source project, then feel free to write an issue.
```
Comment (MargretRensing) [inline CONTRIBUTING.md]:
```suggestion
If you have questions about one of our open-source projects, then feel free to write an issue.
```

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+If this is a general question that doesn't relate to an open source project, then write us a mail at [github@surplex.com](mailto:github@surplex.com).
```
Comment (MargretRensing) [inline CONTRIBUTING.md]:
```suggestion
If this is a general question that doesn't relate to an open source project, then write us a mail to [github@surplex.com](mailto:github@surplex.com).
```

**Review by MargretRensing — COMMENTED** [review]
👍

**Review by jeboehm — APPROVED** [review]

**Pull request merged**